src/progressive-profiling/ProgressiveProfiling.jsx

```diff
@@ -116,20 +116,19 @@ const ProgressiveProfiling = (props) => {
   useEffect(() => {
     if (registrationResult.redirectUrl && authenticatedUser?.userId) {
       const redirectQueryParams = getAllPossibleQueryParams(registrationResult.redirectUrl);
-      if (enablePersonalizedRecommendations && !('enrollment_action' in redirectQueryParams)) {
+      if (enablePopularAndTrendingRecommendations && !('enrollment_action' in redirectQueryParams) && !queryParams?.next) {
```

Comment:
Can't we get the next param from `redirectQueryParams`?

Reply:
No, we can't get the next param from `redirectQueryParams` because when we provide URL to `getAllPossibleQueryParams` it only parses the query params of the provided URL and we are setting the next URL into `registrationResult.redirectUrl` that itself is the next param value.